ghost/core/core/app.js

```diff
@@ -27,12 +27,30 @@ const maintenanceMiddleware = function maintenanceMiddleware(req, res, next) {
     fs.createReadStream(path.resolve(__dirname, './server/views/maintenance.html')).pipe(res);
 };
 
+// Used by Ghost (Pro) to ensure that requests cannot be served by the wrong site
+const siteIdMiddleware = function siteIdMiddleware(req, res, next) {
+    const siteId = config.get('hostSettings:siteId');
+
+    if (req.headers['x-site-id'] === siteId) {
+        return next();
+    }
```

Comment:
I think we should cast both items to a string, so we can be sure of the types. The config is a string (although the ID is a number) but I think we pass the `x-site-id` as a number, so it's just safer to convert to the same

```suggestion
    const siteIdConfig = config.get('hostSettings:siteId');
    const siteIdHeader = req.headers['x-site-id'];

    if (`${siteIdConfig}` === `${siteIdHeader}`) {
        return next();
    }
```